src/others/components/map/Map.tsx

```diff
@@ -1,15 +1,19 @@
 import { useTranslation } from "react-i18next";
-import MapComponent, { Popup, MapRef, MapLayerMouseEvent } from "react-map-gl";
+import MapComponent, { Popup, MapRef, MapLayerMouseEvent, Layer, Source } from "react-map-gl";
 import maplibregl from "maplibre-gl";
 import "maplibre-gl/dist/maplibre-gl.css";
 
 import { Box, Typography } from "@mui/material";
-import { ReactNode, useCallback, useState, useRef } from "react";
-import { MaxRegionVisibleZoomLevel } from "./RegionsSourceWithLayers";
+import { useCallback, useState, useRef, useMemo } from "react";
+import { MaxRegionVisibleZoomLevel, RegionsSourceWithLayers } from "./RegionsSourceWithLayers";
+import { layerStyle } from "./CircleLayerStyle";
+import { layerStyleWithRegions } from "./CircleLayerStyleWithRegions";
+import { RequestMapDataPoint, mapAidRequestsToFeatures } from "../../helpers/map-utils";
+import { FeatureCollection, GeoJsonProperties, Geometry } from "geojson";
+import { useDictionaryContext } from "../../contexts/dictionary-context";
 
 interface MapProps {
-  sourceWithLayer?: ReactNode;
-  interactiveLayerIds: string[]
+  requestMapDataPoints: RequestMapDataPoint[];
```

Comment:
**thought/non-blocking**
If we don't want to spill the map-business logic outside map component, maybe we should rely on a universal data model here (`AidRequest[]`) and do the `AidRequest[]` => `RequestMapDataPoint[]` here. 

So, the interface would accept `AidRequest[]`.

Reply:
Yep!

Reply:
So that `RequestDataPoint` could be mapped to `RequestMapDataPoint` and `RequestTableDataPoint`

Reply:
it should probably not have `Request` in the name either, since for Inventory page it does not make sense.
Will leave this rename for a separate change.